plugins/services/src/js/pages/task-details/TaskFileViewer.js

```diff
@@ -44,22 +44,33 @@ class TaskFileViewer extends React.Component {
       return;
     }
 
-    const routePath = RouterUtil.reconstructPathFromRoutes(routes);
+    let routePath = RouterUtil.reconstructPathFromRoutes(routes);
+    const hasFilePathParam = routePath.endsWith(':filePath');
+    if (!hasFilePathParam && routePath.endsWith('/')) {
+      routePath += ':filePath';
+    }
+    if (!hasFilePathParam && !routePath.endsWith('/')) {
+      routePath += '/:filePath';
+    }
     this.context.router.push(formatPattern(
       routePath,
       Object.assign({}, params, {filePath: encodeURIComponent(path)})
     ));
   }
 
   getLogFiles() {
-    const {directory} = this.props;
+    const {directory, limitLogFiles} = this.props;
     const logViews = [];
     if (!directory) {
       return logViews;
     }
 
     directory.getItems().forEach((item) => {
-      if (!item.isLogFile()) {
+      const excludeFile = (
+        limitLogFiles.length > 0 &&
```

Comment:
let's pre-calculate `limitLogFiles.length > 0` outside the `forEach` cycle as it doesn't change